### PR TITLE
Hooks refactor

### PR DESCRIPTION
**Pull request by lucascomamala**
- Refactored App and Calculator class components to functional components
- Used hook to manage state in Calculator component

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,41 +1,40 @@
...
+const Calculator = () => {
+  // constructor(props) {
+  //   super(props);
+  //   this.state = {
+  //     next: null,
+  //     operation: null,
+  //     total: null,
+  //   };
+  //   this.clickHandler = this.clickHandler.bind(this);
+  // }
```
Comment (AnselemOdims) [inline src/components/Calculator.js]: - [x] Please kindly remove all comments from your codebase as it pollutes the codebase

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @lucascomamala , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Calculator.js (hunk 1 above)
Comment (lucascomamala) [reply src/components/Calculator.js]: Whoops 😅

Comment (cynthiainga) [reply src/components/Calculator.js]: - [x] Please as mentioned by the previous reviewer kindly remove the comments line of code to have a clean codebase, if you think you'll need these codes you will need just to check your commits 🙃 🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @lucascomamala, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.



 ## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct gitflow ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
## Hello @lucascomamala 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**